RecoLocalTracker/SiPixelRecHits/plugins/SiPixelRecHitSoAFromLegacy.cc

```diff
@@ -236,11 +236,16 @@ void SiPixelRecHitSoAFromLegacy::produce(edm::StreamID streamID, edm::Event& iEv
   assert(numberOfHits == numberOfClusters);
 
   // fill data structure to support CA
-  for (auto i = 0; i < 11; ++i) {
+  for (auto i = 0U; i < phase1PixelTopology::numberOfLayers + 1; ++i) {
     output->hitsLayerStart()[i] = hitsModuleStart[cpeView.layerGeometry().layerStart[i]];
   }
-  cms::cuda::fillManyFromVector(
-      output->phiBinner(), 10, output->iphi(), output->hitsLayerStart(), numberOfHits, 256, nullptr);
+  cms::cuda::fillManyFromVector(output->phiBinner(),
+                                phase1PixelTopology::numberOfLayers,
+                                output->iphi(),
+                                output->hitsLayerStart(),
+                                numberOfHits,
+                                256,
```

Comment:
256 should probably be a declared constant.

Reply:
Yes, but not just here. 256 as the number of threads is hardcoded in many places.
Should this be added as an issue to #32483 ? @fwyzard @mmusich @VinInn

Reply:
I'd say that this is a more general issue, that applies to all CUDA kernels, not just the Pixel ones

Reply:
When/if anybody will come with a good reason to make configurable it will be done.  Until then no reason to name it.